mysql/userstat/userstat.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2015 Square, Inc
+//
+
+package userstat
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/square/inspect/metrics"
+	"github.com/square/inspect/mysql/tools"
+	"github.com/square/inspect/os/misc"
```

Comment:
I think we should move the "misc" out of "os" and to util package that can probably also host the callbymethodname function that is defined here.